infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/KubernetesEnvironmentValidatorTest.java

```diff
@@ -51,7 +52,7 @@ public void setUp() throws Exception {
       expectedExceptionsMessageRegExp = "Environment should contain at least 1 pod")
   public void shouldThrowExceptionWhenEnvDoesNotHaveAnyPods() throws Exception {
     // given
-    when(kubernetesEnvironment.getPods()).thenReturn(emptyMap());
+    when(kubernetesEnvironment.getPodsData()).thenReturn(emptyMap());
```

Comment:
Sorry, I'm not sure what you mean. Like when `getPodsData()` is not empty but the actual `pods` and `deployments` are? I think this would be a test case on KubernetesEnvironment directly, since modifying `podData` is / should be impossible (it's only populated from pods and deployments that are in the recipe).

Reply:
yes, I meant a case described by you. I agree that it is rather a case for the environment object itself.